go/internal/flag/flag.go

```diff
@@ -106,8 +106,8 @@ func ParseFlagsForTest() {
 	}
 
 	// parse remaining flags including the log-related ones like --alsologtostderr
-	fs := flag.NewFlagSet("test", flag.ExitOnError)
-	Parse(fs)
```

Comment:
@rohit-nayak-ps Was there a reason that we were creating a new flag set called `test`. If we use a new flag set, then we don't register the flags that are defined in the init function of the tests because they are defined on the CommandLine flagset

Reply:
I guess the go test files I was testing with didn't have flags defined in init functions. Do the tests accept flags like `--alsologtostderr` with your change?

Reply:
Just checked, that they do - 
```
~/vitess   evalengine-integration-test-flags  go test -v -run="^TestGenerateFuzzCases$" ~/vitess/go/vt/vtgate/evalengine/integration -args --fuzz-total=0 --alsologtostderr
I0928 17:21:10.412698   91189 local_cluster.go:292] No environment in cluster settings. Creating default...
I0928 17:21:10.415022   91189 local_cluster.go:299] LocalCluster environment: &{BasePort:11791 TmpPath:/Users/manangupta/vitess/vtdataroot/vttest1473485283 DefaultMyCnf:[/Users/manangupta/vitess/config/mycnf/test-suite.cnf] Env:[VTDATAROOT=/Users/manangupta/vitess/vtdataroot/vttest1473485283 MYSQL_FLAVOR=MySQL56]}
I0928 17:21:10.415169   91189 local_cluster.go:322] Initializing MySQL Manager (*vttest.Mysqlctl)...
I0928 17:21:15.611991   91189 local_cluster.go:498] Creating databases in cluster...
I0928 17:21:15.612903   91189 local_cluster.go:529] Execute(): "create database `vttest`"
I0928 17:21:15.614614   91189 local_cluster.go:346] MySQL up: {"host":"","port":0,"uname":"vt_dba","pass":"","dbname":"","unix_socket":"/Users/manangupta/vitess/vtdataroot/vttest1473485283/vt_0000000001/mysql.sock","charset":"","flags":0,"ssl_mode":"","ssl_ca":"","ssl_ca_path":"","ssl_cert":"","ssl_crl":"","ssl_key":"","tls_min_version":"","server_name":"","connect_timeout_ms":0,"DeprecatedDBName":"","DisableClientDeprecateEOF":false,"EnableQueryInfo":false}
I0928 17:21:15.614626   91189 local_cluster.go:358] Mysql data directory does not exist. Initializing cluster with database and vschema migrations...
=== RUN   TestGenerateFuzzCases
    fuzz_test.go:163: skipping fuzz test generation
--- SKIP: TestGenerateFuzzCases (0.00s)
PASS
ok      vitess.io/vitess/go/vt/vtgate/evalengine/integration    7.092s
 ~/vitess   evalengine-integration-test-flags  
```

I had added the same flag set